dotnet/src/webdriver/SeleniumManager.cs

```diff
@@ -0,0 +1,146 @@
+using OpenQA.Selenium.Internal;
+
+namespace OpenQA.Selenium
+{
+    /// <summary>
+    /// Wrapper for the Selenium Manager binary.
+    /// </summary>
+    public static class SeleniumManager
+    {
+        private static string binary;
+        private static readonly List<string> KnownDrivers = new List<string>() {
+            "geckodriver.exe",
+            "chromedriver.exe"
+        };
+
+        /// <summary>
+        /// Determines the location of the correct driver.
+        /// </summary>
+        /// <param name="driverName">Which driver the service needs.</param>
+        /// <returns>
+        /// The location of the driver.
+        /// </returns>
+        public static string DriverPath(string driverName)
+        {
+            if (!KnownDrivers.Contains(driverName))
+            {
+                throw new WebDriverException("Unable to locate driver with name: " + driverName);
+            }
+            var binaryFile = Binary;
+            if (binaryFile == null) return null;
+
+            var arguments = "--driver " + driverName.Replace(".exe", "");
+            var output = RunCommand(binaryFile, arguments);
+            return output.Replace("INFO\t", "").TrimEnd();
+        }
+
+        /// <summary>
+        /// Gets the location of the correct Selenium Manager binary.
+        /// </summary>
+        private static string Binary
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(binary))
+                {
+                    string folder = "windows";
+                    string extension = ".exe";
+
+                    if (!Environment.OSVersion.Platform.ToString().StartsWith("Win"))
+                    {
+                        throw new WebDriverException("Selenium Manager only supports Windows in .NET at this time");
+                    }
+
+                    try
+                    {
+                        string name = "selenium-manager-" + folder;
+                        using (Stream fileStream = ResourceUtilities.GetResourceStream(name, name))
+                        {
+                            using (BinaryReader binReader = new BinaryReader(fileStream, Encoding.ASCII))
+                            {
+                                byte[] fileBytes = binReader.ReadBytes((int)fileStream.Length);
+                                string directoryName = string.Format(CultureInfo.InvariantCulture, "webdriver.{0}",
+                                    Guid.NewGuid().ToString("N"));
+                                var path = Path.Combine(Path.GetTempPath(), directoryName + "/" + folder);
+                                Directory.CreateDirectory(path);
+                                var filePath = Path.Combine(path, "selenium-manager" + extension);
+
+                                using (BinaryWriter binWriter = new BinaryWriter(File.Open(filePath, FileMode.Create)))
+                                {
+                                    binWriter.Flush();
```

Comment:
Redundant invocation, `Flush` is automatically used when disposing.

Reply:
but the flush is before the writing not after (with the disposal). tbh I'm not sure why you need it, I was going off of what I saw someone else doing...